WooCommerce/WooCommerceTests/ViewRelated/Shipping Label/Create Shipping Label/ShippingLabelCustomPackageFormViewModelTests.swift

```diff
@@ -18,4 +19,239 @@ class ShippingLabelCustomPackageFormViewModelTests: XCTestCase {
         XCTAssertEqual(viewModel.emptyPackageWeight, "")
     }
 
+    func test_package_name_validation_succeeds_if_name_is_not_empty() {
+        // Given
+        let viewModel = ShippingLabelCustomPackageFormViewModel()
+
+        // When
+        viewModel.packageName = "Test"
+
+        // Then
+        let _ = viewModel.packageNameValidation.sink { validated in
+            XCTAssertTrue(validated)
+        }
```

Comment:
Although this works, the test can fail silently if `packageNameValidation` doesn't emit any event. Which means the block within `.sink` will not be triggered and there will be no assertion at all.

When testing with asynchronous code, it is recommended that we use `expectation` and wait for timeout to make sure that we catch the error if a publisher doesn't emit the expected event. I'd recommend this [nice article](https://www.swiftbysundell.com/articles/unit-testing-combine-based-swift-code/) as a guide for testing Combine code.

However, if we use `@Published` for validation instead, we can assert the value of the variable directly - and therefore there's no need for using `.sink` and `expectation`.